libshaderc_util/include/libshaderc_util/spirv_tools_wrapper.h

```diff
@@ -34,6 +34,22 @@ bool SpirvToolsAssemble(const string_piece assembly, spv_binary* binary,
 // *text_or_error.
 bool SpirvToolsDisassemble(const std::vector<uint32_t>& binary,
                            std::string* text_or_error);
+
+// Used to tell SpirvToolsOptimize() whether to enable a certain pass.
+//
+// TODO(antiagainst): This may need to consider pass order once we have the
+// interface ready for users to specify optimization on a per pass base.
+// But now let's be simple.
+struct EnableOptimizationPass {
```

Comment:
This is too weak an abstraction.  
It's important to get this right because it's at the API surface.

We have two choices:
1) The SpirvToolsOptimize interface should use the same "levels" of Zero, Size and have that interpreted inside into the more fine grain passes.  Or
2) The SpirvToolsOptimize interface should take a list of the opaque "transform" objects exposed by the spirv-tools optimize.hpp.   Then EnableOptimizationPass should just be vector of those things.  
The upside of 2 is flexibility, the downside is you now expose the interior API details but that's ok because "SpirvTools*" is in the name of the function!  The other extra complexity is that you now need to track *which* object is the strip-debug-info transform object.  But I'm sure there's several ways to solve that.